File: Daily/sbgo/Leetcode414-第三大的数.go
package sbgo

// golang的int是根据os变的，64位os就是64位
// 所以在做的时候可能没有影响，但是还是应该设成int64

func thirdMax(nums []int) int {
	INF := (int)(-1e10)
	cnt, f1, f2, f3 := 0, INF, INF, INF
	for _, x := range nums {
		if x > f3 {
			cnt++
			f1, f2, f3 = f2, f3, x
		} else if x < f3 && x > f2 {
			cnt++
			f1, f2 = f2, x
		} else if x < f2 && x > f1 {
			cnt++
			f1 = x
		}
	}
	if cnt < 3 {
		return f3
	}
	return f1
}

File: Daily/PY/Leetcode414-第三大的数.py
from typing import List
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        cnt = 0
        f1 = f2 = f3 = int(-1e10)
        for i in nums:
            if i > f3: 
                cnt += 1
                f1,f2,f3 = f2,f3,i
            elif i < f3 and i > f2:
                cnt += 1
                f1,f2 = f2,i
            elif i < f2 and i > f1:
                cnt += 1
                f1 = i
        res = f3 if cnt < 3 else f1
        return res